loopy/symbolic.py

```diff
@@ -1013,11 +1036,12 @@ def __init__(self, rule_mapping_context, subst_func, within):
         self.within = within
 
     def map_variable(self, expr, expn_state):
-        if (expr.name in expn_state.arg_context
-                or not self.within(
-                    expn_state.kernel, expn_state.instruction, expn_state.stack)):
-            return super().map_variable(
-                    expr, expn_state)
+        if expn_state.instruction is not None:
```

Comment:
Why aren't we calling the superclass in this case? (Add a comment to explain, if there's a good reason.)

Reply:
Changed the code slightly to make is more explainable. 5737a4c

Reply:
Got rid of these changes. So this is comment is no longer valid. (Resolving it).